### PR TITLE
dasdfmt cleanup

### PR DESCRIPTION
**Pull request by hreinecke**
Hi all,

here's a small patchset to cleanup some issues with dasdfmt; I have another patchset allowing to specify several devices to dasdfmt for which this patchset is prerequisite.

### CODE REVIEW

**Comment by michael-holzheu** [comment]
@hreinecke : IMHO the first commit makes sense to me. But since it has nothing to do with the dasdfmt cleanup, could you please open a second pull request for that one?
@hoeppnerj : Could you have a look at the three other patches?

File: dasdfmt/dasdfmt.c [hunk 1]
```diff
@@ -527,9 +527,9 @@ static void get_device_info(dasdfmt_info_t *info)
 /*
  * Retrieve blocksize of device
  */
-static void get_blocksize(unsigned int *blksize)
+static void get_blocksize(int fd, unsigned int *blksize)
```
Comment (hoeppnerj) [inline dasdfmt/dasdfmt.c]: There is no need to introduce a local file descriptor here. Just use the global one.

File: dasdfmt/dasdfmt.c [hunk 2]
```diff
@@ -970,28 +970,19 @@ static void dasdfmt_print_info(dasdfmt_info_t *info, volume_label_t *vlabel,
...
-	f = open(info->devname, O_RDONLY);
-	if (f == -1)
-		ERRMSG_EXIT(EXIT_FAILURE, "%s: Unable to open device %s: %s\n",
-			    prog_name, info->devname, strerror(errno));
-
-	get_blocksize(&blksize);
-
-	if (close(f) != 0)
-		ERRMSG("%s: error during close: %s\ncontinuing...\n",
-		       prog_name, strerror(errno));
```
Comment (hoeppnerj) [inline dasdfmt/dasdfmt.c]: I cleaned up bits here before, I seem to have forgotten to remove the file handling here. Looks good, thanks!

File: dasdfmt/dasdfmt.c [hunk 3]
```diff
@@ -970,28 +970,19 @@ static void dasdfmt_print_info(dasdfmt_info_t *info, volume_label_t *vlabel,
...
+	get_blocksize(fd, &blksize);
 
-	if ((strncmp(info->dasd_info.type, "ECKD", 4) == 0) &&
-	    (!info->dasd_info.FBA_layout)) {
+	if ((strncmp(dasd_info->type, "ECKD", 4) == 0) &&
+	    (!dasd_info->FBA_layout)) {
```
Comment (hoeppnerj) [inline dasdfmt/dasdfmt.c]: You can drop the parentheses around 'dasd_info->FBA_layout' while you're at it.

File: dasdfmt/dasdfmt.c [hunk 4]
```diff
@@ -970,28 +970,19 @@ static void dasdfmt_print_info(dasdfmt_info_t *info, volume_label_t *vlabel,
 /*
  * get volser
  */
-static int dasdfmt_get_volser(dasdfmt_info_t *info, char *volser)
+static int dasdfmt_get_volser(int fd, char *devname,
```
Comment (hoeppnerj) [inline dasdfmt/dasdfmt.c]: As a result, same here. No local fd needed.

**Comment by hreinecke** [comment]
I've created a new pull request addressing the mentioned issues, so this one can be closed.

**Pull request closed without merging**